src/uu/shuf/src/shuf.rs

```diff
@@ -10,132 +10,165 @@
 #[macro_use]
 extern crate uucore;
 
+use clap::{App, Arg};
 use rand::Rng;
 use std::fs::File;
 use std::io::{stdin, stdout, BufReader, BufWriter, Read, Write};
-use std::usize::MAX as MAX_USIZE;
+
+struct Options {
+    head_count: usize,
+    output: Option<String>,
+    random_source: Option<String>,
+    repeat: bool,
+    sep: u8,
+}
 
 enum Mode {
-    Default,
-    Echo,
+    Default(String),
+    Echo(Vec<String>),
     InputRange((usize, usize)),
 }
 
 static NAME: &str = "shuf";
 static VERSION: &str = env!("CARGO_PKG_VERSION");
+static OPT_ECHO: &str = "echo";
+static OPT_INPUT_RANGE: &str = "input-range";
+static OPT_HEAD_COUNT: &str = "head-count";
+static OPT_OUTPUT: &str = "output";
+static OPT_RANDOM_SOURCE: &str = "random-source";
+static OPT_REPEAT: &str = "repeat";
```

Comment:
Please use an options mod for this:
https://github.com/uutils/coreutils/blob/83f8140aafe80a9683eb008362caedef87edd831/src/uu/cat/src/cat.rs#L129-L140

Reply:
Sure! Thanks.